packages/svelte/src/internal/client/runtime.js

```diff
@@ -379,13 +379,15 @@ function execute_signal_fn(signal) {
 				// If we have more than 16 elements in the array then use a Set for faster performance
 				// TODO: evaluate if we should always just use a Set or not here?
 				const full_current_dependencies_set =
-					current_dep_length > 16 ? new Set(full_current_dependencies) : null;
+					current_dep_length > 16 && deps_length - current_dependencies_index > 1
+						? new Set(full_current_dependencies)
+						: null;
 				for (i = current_dependencies_index; i < deps_length; i++) {
 					const dependency = dependencies[i];
 					if (
-						(full_current_dependencies_set !== null &&
-							!full_current_dependencies_set.has(dependency)) ||
-						!full_current_dependencies.includes(dependency)
+						full_current_dependencies_set !== null
+							? !full_current_dependencies_set.has(dependency)
+							: !full_current_dependencies.includes(dependency)
```

Comment:
this always ran both when the dependency wasn't in the set previously, slowing things down.
I don't know if this was profiled, but if it was and it didn't appear that this was slower, I'm wondering if this kind of perf optimization is worth it in general